homeassistant/components/light/c_by_ge.py

```diff
@@ -0,0 +1,129 @@
+"""
+Support for C by GE lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.c_by_ge/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME, CONF_PASSWORD
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, ATTR_COLOR_TEMP, SUPPORT_COLOR_TEMP,
+    Light, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.color import \
+    color_temperature_kelvin_to_mired as kelvin_to_mired
+from homeassistant.util.color import \
+    color_temperature_mired_to_kelvin as mired_to_kelvin
+
+REQUIREMENTS = ['laurel==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities,
+                   discovery_info=None):
+    """Set up the C by GE platform."""
+    import laurel
+
+    data = laurel.laurel(config[CONF_NAME], config[CONF_PASSWORD])
+    for network in data.networks:
+        network.connect()
+
+    lights = []
+    for bulb in data.devices:
+        lights.append(GELight(bulb))
+
+    add_entities(lights)
+
+
+class GELight(Light):
+    """Representation of a C by GE light."""
+
+    def __init__(self, bulb):
+        """Initialize the light."""
+        self._bulb = bulb
+        self._state = False
+        self._brightness = 255
+        bulb.set_callback(self.callback, None)
+
+        if bulb.supports_temperature():
+            self._features = SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+        else:
+            self._features = SUPPORT_BRIGHTNESS
+
+    def callback(self, args):
+        """Handle updates from the bulb."""
+        if self.hass is not None:
```

Comment:
Remove this check. It's not necessary.

Reply:
If the callback is called before `async_added_to_hass` then `schedule_update_ha_state` will fail because `self.hass` is not set yet (had a similar issue recently)